### PR TITLE
Feature/document resources

### PR DESCRIPTION
**Pull request by boralyl**
This will create a section named 'Resource Objects' at the end of the
api documentation that documents all objects used in requests/responses.

It will also add an inline reference link to the documentation next to
the parameter in the api documentation.

_**Example of the auto added inline reference:**_

screen shot 2018-05-15 at 4 10 13 pm

_**Example of the linked reference that documents the object:**_

screen shot 2018-05-15 at 4 18 46 pm

### CODE REVIEW

File: doctor/docs/base.py [hunk 1]
```diff
@@ -251,6 +274,51 @@ def get_json_lines(annotation: ResourceAnnotation, field: str, route: str,
...
+    """Generate documentation lines for all collected resource objects.
+
+    As API documentation is generated we keep a running list of objects used
+    in request parameters and responses.  This section will generate
+    documentation for each object and provide an inline reference in the API
+    documentation.
+
+    :returns: A list of lines required to generate the documentation.
+    """
+    lines = ['Resource Objects', '----------------']
```
Comment (boralyl) [inline doctor/docs/base.py]: Thought's on this name for the heading?  I don't hate it, but don't love it either.

**Review by pkar — APPROVED** [review]

File: doctor/docs/base.py [hunk 2]
```diff
@@ -295,6 +363,19 @@ def prefix_lines(lines, prefix):
...
+def class_name_to_resource_name(class_name: str) -> str:
+    """Converts a camel case class name to a resource name with spaces.
+
+    >>> class_name_to_resource_name('FooBarObject')
+    'Foo Bar Object'
+
+    :param class_name: The name to convert.
+    :returns: The resource name.
+    """
+    s = re.sub('(.)([A-Z][a-z]+)', r'\1 \2', class_name)
```
Comment (noonat) [inline doctor/docs/base.py:385]: What case are you expecting this substitution to handle? It looks like the substitution below would handle things, and when I ran the examples from your test case through that, all of them were converted correctly.

File: doctor/docs/base.py (hunk 1 above)
Comment (noonat) [reply doctor/docs/base.py]: I'm also ambivalent about it, but I also can't think of anything better so I think we can just stick with this.

File: doctor/docs/base.py (hunk 2 above)
Comment (boralyl) [reply doctor/docs/base.py:385]: I forgot to add a test for this case.  This ensures if the class names starts with an acronym in all caps that it splits properly.

e.g. 

```python
>>> class_name_to_resource_name('HTTPResponse')
'HTTP Response'
```

Without the first regex it wouldn't alter the name at all and would just be `HTTPResponse`.

**Pull request merged**